From 68dbf16886540b4ec6edc46409f5adf0b97f1dcf Mon Sep 17 00:00:00 2001
From: "Dr. David Alan Gilbert (git)" <dgilbert@redhat.com>
Date: Mon, 20 Oct 2014 14:55:55 +0200
Subject: [PATCH 43/43] rdma: Fix block during rdma migration

Message-id: <1413816955-4190-2-git-send-email-dgilbert@redhat.com>
Patchwork-id: 61746
O-Subject: [RHEL-7.1 qemu-kvm PATCH 1/1] rdma: Fix block during rdma migration
Bugzilla: 1152969
RH-Acked-by: Juan Quintela <quintela@redhat.com>
RH-Acked-by: Amit Shah <amit.shah@redhat.com>
RH-Acked-by: Marcel Apfelbaum <marcel.a@redhat.com>

From: Gonglei <arei.gonglei@huawei.com>

If the networking break or there's something wrong with rdma
device(ib0 with no IP) during rdma migration, the main_loop of
qemu will be blocked in rdma_destroy_id. I add rdma_ack_cm_event
to fix this bug.

Signed-off-by: Mo Yuxiang <Moyuxiang@huawei.com>
Signed-off-by: Gonglei <arei.gonglei@huawei.com>
Reviewed-by: Michael R. Hines <mrhines@us.ibm.com>
Signed-off-by: Juan Quintela <quintela@redhat.com>
(cherry picked from commit 2a9343470432573acf06379bcbf8e5e6568507e4)
---
 migration-rdma.c | 1 +
 1 file changed, 1 insertion(+)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 migration-rdma.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/migration-rdma.c b/migration-rdma.c
index ba4ad76..5596571 100644
--- a/migration-rdma.c
+++ b/migration-rdma.c
@@ -949,6 +949,7 @@ route:
         ERROR(errp, "result not equal to event_addr_resolved %s",
                 rdma_event_str(cm_event->event));
         perror("rdma_resolve_addr");
+        rdma_ack_cm_event(cm_event);
         ret = -EINVAL;
         goto err_resolve_get_addr;
     }
-- 
1.8.3.1

